crates/sui/src/unit_tests/cli_tests.rs

```diff
@@ -767,6 +767,47 @@ async fn test_package_publish_command_failure_invalid() -> Result<(), anyhow::Er
     Ok(())
 }
 
+#[sim_test]
+async fn test_package_publish_nonexistent_dependency() -> Result<(), anyhow::Error> {
+    let mut test_cluster = TestClusterBuilder::new().build().await?;
+    let address = test_cluster.get_address_0();
+    let context = &mut test_cluster.wallet;
+
+    let client = context.get_client().await?;
+    let object_refs = client
+        .read_api()
+        .get_owned_objects(
+            address,
+            Some(SuiObjectDataOptions::full_content()),
+            None,
+            None,
+            None,
+        )
+        .await?
+        .data;
+
+    let gas_obj_id = object_refs.first().unwrap().object().unwrap().object_id;
+
+    let mut package_path = PathBuf::from(TEST_DATA_DIR);
+    package_path.push("module_publish_with_nonexistent_dependency");
+    let build_config = BuildConfig::default();
+    let result = SuiClientCommands::Publish {
+        package_path,
+        build_config,
+        gas: Some(gas_obj_id),
+        gas_budget: 20_000,
+        skip_dependency_verification: false,
+        with_unpublished_dependencies: true,
+    }
+    .execute(context)
+    .await;
+
+    assert!(
+        std::string::ToString::to_string(&result.unwrap_err()).contains("DependentPackageNotFound")
+    );
```

Comment:
not a huge fan of this assert, but it's what the analogous `authority_test.rs` does, so I can live with it. The actual error for expect testing is more informative, but runs into the problem of needing to sanitize addrs:

```elixir
Err(
    RpcError(
        Call(
            Custom(
                ErrorObject {
                    code: ServerError(
                        -32000,
                    ),
                    message: "Transaction has non recoverable errors from at least 1/3 of validators: [(UserInputError { error: DependentPackageNotFound { package_id: 0x0000000000000000000000000000000000000000000000000000000000abc123 } }, [k#8cbb4f312ea3d96cfc1df1f5ca0a0dbc5c114922a3b9c763d5547ff4c2b5bcaea958699f67f0e25020be5f39ec5245b50efbc11090a6028f9eb91dec80d2fcb8a16929d6f16088c231339f3cd5321ac1e172c85cc94b1c305b5986eb627b4382], 2500), (UserInputError { error: DependentPackageNotFound { package_id: 0x0000000000000000000000000000000000000000000000000000000000abc123 } }, [k#a46530e71812e7c29e02374f3ed10c277a654df9dcb216e13a380f854ef599364eba7b2041c04d59da7faf89882cae7e0895c05de08c6ce3fd8774b979ad9609e233b53a632df7de69bc047825c1799061f2760dea795e8681c19b1d5b815167], 2500), (UserInputError { error: DependentPackageNotFound { package_id: 0x0000000000000000000000000000000000000000000000000000000000abc123 } }, [k#91a96f6b0cd961e01f4578d0d6dfea9c0636f23cb0d3fc6e8ebc497652cd9abaaf3af185c861a3a27da570051968e82a06c0021b7da65bf11423bc6d09ad0e53cc78a6e0605c599aa7344646d2b3e3714eed59fe5c33d6dc5aabe4e9a4bc38eb], 2500)].",
                    data: None,
                },
            ),
        ),
    ),
)
```

Reply:
Yeah, I think it's fine -- tough to get something with a good structure out from the RPC Errors